apollo-composition/src/lib.rs

```diff
@@ -355,3 +374,35 @@ fn satisfiability_result_into_issues(
         Err(issue) => Either::Right(once(issue)),
     }
 }
+
+impl From<BuildError> for Issue {
+    fn from(error: BuildError) -> Issue {
+        Issue {
+            code: error.code.unwrap_or("UNKNOWN_ERROR_CODE".to_string()),
+            message: error.message.unwrap_or("Unknown error".to_string()),
```

Comment:
```suggestion
            code: error.code.unwrap_or_else(|| "UNKNOWN_ERROR_CODE".to_string()),
            message: error.message.unwrap_or_else(|| "Unknown error".to_string()),
```

To avoid allocating those strings unless the error was actually missing details